zipline/algorithm.py

```diff
@@ -287,11 +287,16 @@ def __init__(self, *args, **kwargs):
             )
 
         # If a schedule has been provided, pop it. Otherwise, use NYSE.
-        self.trading_calendar = kwargs.pop(
+        calendar = kwargs.pop(
             'trading_calendar',
-            get_calendar("NYSE")
+            "NYSE"
         )
 
+        if not calendar:
+            self.trading_calendar = get_calendar("NYSE")
+        else:
+            self.trading_calendar = get_calendar(calendar)
+
```

Comment:
I'm pretty sure that if I'm using `kwargs.pop()` above that I can remove the `if` statement and just assign `self.trading_calendar` to `get_calendar(calendar)`

Reply:
Yup, you've encapsulated the logic of the conditional statement in the `pop`: if `trading_calendar` is not in the `kwargs` then use `"NYSE"`.